src/state_point.F90

```diff
@@ -223,7 +240,28 @@ subroutine write_state_point()
             filter_group = create_group(tally_group, "filter " // &
                  trim(to_str(j)))
 
-            call write_dataset(filter_group, "type", tally%filters(j)%type)
+            ! Write name of type
+            select case (tally%filters(j)%type)
```

Comment:
That is very true. I may do a branch in the near future cleaning up things like this.